### DISCUSSION TITLE
still active?

### DISCUSSION THREAD

**Discussion by gustedt** [thradams/cake#30] [General]
Hi,
are you still active in developping this?

I see that there are some more or less obvious bugs that have that this is not really possible to use beyond small examples. 

C23 will come out at the end of the year, I was wondering if cake could be an alternative to propose to people then, if they are stuck with compilers that still struggle to implement certain things.

Thanks
Jens

**Comment by thradams** [comment]
Hi,

Although I am not working on it regularly it still alive. (This is a side project)

Definitely the proximity of C23 is a motivation! 

Some missing features are general (like better semantic analysis) and others are more related with C23 transpilation.

For instance, (bug)

```
auto s = "hello auto";
```
generates 
```c
const char   s[11] = "hello auto";
```
instead of
```c
const char  *s = "hello auto";
````

Some features of C23, such as static in the compound literal, are missing. 
Feel free to open bugs and send suggestions that will be very welcome.

Did you see more bugs in existing implementation?

Some items I know are missing or with problem: 

*  embed is not 100% implemented
* static compound literals
*  C23 enum additions (I guess I am parsing but not converting checking..)
* Remove all attributes when comping to <= C11 
* nullptr semantics are not implemented (just keyword and conversion to ((void*)0)
*

**Comment by gustedt** [comment]
Hi,
yes, this seems a general problem that the initializer of an auto variable is not undergoing the requested conversions: lvalue conversion, array-to-pointer and function-to-pointer.

Ok, I'll post the bugs I find.
Thanks for doing this
Jens
